### ISSUE TITLE
GDScript 2.0: Calling an overwritten function of an external inner class results in a runtime error

### ISSUE THREAD

**Issue by MikeSchulze** [godotengine/godot#67977]
### Godot version

v4.0.beta3.official [01ae26d31]

### System information

Windows 10

### Issue description

I have a mocker that stubs classes
On Godot 4 it is broken when calling an  overwritten function of an external inner class.

![image](https://user-images.githubusercontent.com/347037/198675043-a749c233-8053-47ff-9294-341ba0527e4a.png)

### Steps to reproduce

run the example code
external.gd
```
class_name External

class A:
	func foo() -> void:
		prints("External.A:foo")
	
	func bar() -> void:
		pass
```

testscene.gd
```
class_name TestScene
extends Node

class A:
	func foo() -> void:
		prints("A:foo")

class B extends A:
	func foo() -> void:
		super.foo()
		prints("B:foo")
	
	func bar() -> void:
		pass

class C extends External.A:
	func foo() -> void:
		super.foo()
		prints("C:foo")
	
	func bar() -> void:
		pass

func _ready():
	prints("External.A")
	var a := External.A.new()
	a.bar()
	a.foo()
	
	prints("B")
	var b := B.new()
	b.bar()
	b.foo()
	prints("C")
	var c := C.new()
	c.bar()
	# fails by a runtime error
	c.foo()
```
The output
```
External.A
External.A:foo
B
A:foo
B:foo
C
  editor/debugger/debug_adapter/debug_adapter_types.h:70 - Condition "path.is_empty()" is true.

```
![image](https://user-images.githubusercontent.com/347037/198677385-63013647-4651-4b0c-8712-efd51359cf21.png)

**Comment by adamscott** [comment]
My PR #67714 now fixes this issue, as my previous fix wouldn't support extending global classes.

@MikeSchulze Thanks for the example code, but it would be better to join an MRP rather than posting the code in code blocks. Easier to setup and test.

**Comment by MikeSchulze** [comment]
> My PR #67714 now fixes this issue, as my previous fix wouldn't support extending global classes.
> 
> @MikeSchulze Thanks for the example code, but it would be better to join an MRP rather than posting the code in code blocks. Easier to setup and test.

No problem, I will do it on the next bugs if I find some more ;)

**Comment by rune-scape** [comment]
@akien-mga just tested that this was fixed by #68374
